src/cpp/src/tokenizer.cpp

```diff
@@ -109,17 +112,46 @@ class Tokenizer::TokenizerImpl {
 
     std::string m_chat_template = {};
 
+    std::pair<int, int> get_padding_values(PaddingMode padding_mode, size_t max_length) {
+        switch (padding_mode) {
+            case PaddingMode::TRUNCATE:
+                return {max_length, 0};
+            case PaddingMode::LONGEST:
+                return {std::numeric_limits<int32_t>::max(), 0};
+            case PaddingMode::MAX_LENGTH:
+                return {std::numeric_limits<int32_t>::max(), max_length};
+            case PaddingMode::DO_NOT_PAD:
+                // behaves exactly as longest
+                // TODO: need to find a way to disable padding automatically so that it will match to HF.
+                return {std::numeric_limits<int32_t>::max(), 0};
+            default:
```

Comment:
NONE is missed here. Do we need NONE at all?

Reply:
Simplified regimes a little bit. There no NONE anymore. Now there is only 3 regimes. Please take a look to the latest changes.